x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx

```diff
@@ -150,6 +152,16 @@ const CreateRulePageComponent: React.FC = () => {
     }),
     [indicesConfig, threatIndicesConfig]
   );
+  const aboutStepDefault = useMemo(() => {
+    const defaultMaxSignals =
+      maxAlertsPerRun !== undefined && maxAlertsPerRun < DEFAULT_MAX_SIGNALS
+        ? maxAlertsPerRun
+        : DEFAULT_MAX_SIGNALS;
+    return {
+      ...stepAboutDefaultValue,
+      maxSignals: defaultMaxSignals,
+    };
+  }, [maxAlertsPerRun]);
```

Comment:
Do we need to implement similar logic on the backend? If I set in the Kibana config `xpack.alerting.rules.run.alerts.max: 5`, the rule edit form will set max signals to 5 by default. However, if I remove the field value and save the rule, it resets to 100 instead of 5. It's not a huge issue, but perhaps it's worth aligning the frontend and backend behavior.

Reply:
Talked more about this logic with Marshall, what do you think of getting rid of the changing default altogether and just having it always default to 100? It does seem like a bit of a corner case for the UI that would be more easily covered with consistent logic in the rule executors themselves (using the min of the two values). Something about having a changing "default" seems confusing imo.

Reply:
Sounds good 👍 Let's keep the default at 100, regardless of Kibana settings. This approach would be best for maintenance.